s3restore.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+var (
+	Version = "Placeholder"
+)
+
+type ParsedArgs struct {
+	CommandName string
+	Args        map[string]string
+}
+
+type S3svc struct {
```

Comment:
I don't think this needs to be a struct, as we are only interested in extending the `*s3.S3` type by attaching methods to it. It is not storing or composing anything. You could just do:

```
type S3svc *s3.S3
```

It would simplify method calls so they are `s.S3svc` instead of `s.S3svc.Svc`.

However, I don't expect you to change it. It can stay as it is.